ide/coqide/coqide_ui.ml

```diff
@@ -160,15 +164,18 @@ let init () =
 \n  </menu>\
 \n</menubar>\
 \n<toolbar name='CoqIDE ToolBar'>\
+\n  <toolitem action='Open' />\
 \n  <toolitem action='Save' />\
 \n  <toolitem action='Close buffer' />\
+\n  <separator />\
 \n  <toolitem action='Forward' />\
 \n  <toolitem action='Backward' />\
 \n  <toolitem action='Run to cursor' />\
+\n  <toolitem action='Reset' />\
```

Comment:
Reset should remain where it was.  It's an infrequent operation.  `Force` should go away,

Reply:
I’m not sure it’s so infrequent. To me, it has an important (and common) role: *clear everything*, i.e. clear the panels, remove the highlight, rest the mind, and overcome bugs in CoqIDE (in my class, some of us had to Reset the script execution because CoqIDE stuck at some point). I think it is psychologically important to quickly know how to reset everything. It helps people trust the program and have hindsight.

But more important is: the order of items must not be determined by habits of experienced users, but by how much it will be easy to understand by new users. I think the order of items doesn’t matter for experienced users: they should rather use hotkeys. If you’re still using buttons and menu items when you’re experienced (except for demonstrating something to somebody), I’m sad for you…! Even if you are, the order doesn’t matter because you know where they are. These GUI components are mostly meant for people who don’t know the commands by heart. Striking a correct balance between (1) ensuring to advanced users that they will always be an efficient way to use the program, and (2) making the program easy to understand by (and, in some way, playful for) new users, is something very important to me. I feel program designers always focus either on (1) or (2), the other point being really forgotten. It tires me and even makes me angry, because it enforces gaps between experienced people and laypeople, which is very bad for society.

I could think about another organization: moving the Interrupt closer to the beginning of the list, e.g.:
Forward | Backward | Run to cursor | Interrupt | Fully check | Run to end | Reset
But I think it is actually good to have Interrupt at the end because it’s something that you will use in a quite negative case (i.e. when computing is too long or turns out in an infinite loop — I don’t know if we can have infinite loops in Coq). So it’s psychologically a good choice to have it at the end; it helps have the mind well-structured.

(About “Force,” see previous reply.)

Reply:
We'll figure something out, but I think there are higher priority things to fix than menus, mnemonics and accelerators.  Some features are unusable, others just don't make much sense.

Reply:
It was quick to fix and to me this is an important base to have before working on more complex things. This is not complex because it is very classical; complexity will be in things which are specific to our piece of software.

I’m not aware of these unusable or inappropriate features…! I will be interested in improving and fixing.